google/cloud/documentai_toolbox/converters/vision_helpers.py

```diff
@@ -0,0 +1,326 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Helper functions for docproto to vision conversion."""
+
+import dataclasses
+from typing import List, Union
+
+import immutabledict
+
+from google.cloud.documentai import Document
+from google.cloud.vision import TextAnnotation, Symbol, Word, Paragraph, Block, Page
+from google.cloud import vision
+
+
+_BREAK_TYPE_MAP = immutabledict.immutabledict(
+    {
+        Document.Page.Token.DetectedBreak.Type.SPACE: (
+            TextAnnotation.DetectedBreak.BreakType.SPACE
+        ),
+        Document.Page.Token.DetectedBreak.Type.WIDE_SPACE: (
+            TextAnnotation.DetectedBreak.BreakType.LINE_BREAK
+        ),
+        Document.Page.Token.DetectedBreak.Type.HYPHEN: (
+            TextAnnotation.DetectedBreak.BreakType.HYPHEN
+        ),
+    }
+)
+
+
+ElementWithLayout = Union[
+    Document.Page.Paragraph,
+    Document.Page,
+    Document.Page.Token,
+    Document.Page.Block,
+    Document.Page.Symbol,
+]
+
+
+@dataclasses.dataclass
+class PageInfo:
+    page: Document.Page
+    text: str = ""
+    page_idx: int = 0
+    block_idx: int = 0
+    paragraph_idx: int = 0
+    token_idx: int = 0
+    symbol_idx: int = 0
+
+
+def _get_text_anchor_substring(text: str, text_anchor: Document.TextAnchor) -> str:
+    """Gets text corresponding to the TextAnchor."""
+    return "".join(
+        [
+            text[text_segment.start_index : text_segment.end_index]
+            for text_segment in text_anchor.text_segments
+        ]
+    )
+
+
+def _has_text_segment(layout: Document.Page.Layout) -> bool:
+    """Checks if the layout has not empty text_segments."""
+    if layout.text_anchor.text_segments:
+        return True
+    return False
```

Comment:
Is there a reason for this function to exist? Seems like it's redundant

Reply:
It looks like this function is used to check if the text_anchor is empty or not. Also it makes it more readable since there isn't a long if statement.

Reply:
Right, it just seems a little verbose. Maybe even just change it to

```
return layout.text_anchor.text_segments
```

Since it will evaluate the boolean